src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace System.Buffers
+{
+    internal sealed class DefaultArrayPool<T> : ArrayPool<T>
+    {
+        private const int MinimiumArraySize = 16;
+        private int _maxArrayLength;
+        private int _numberOfArraysPerBucket;
+        private bool _isTObjectType;    // cache this to only do the reflection call once
```

Comment:
Agreed...but it is better than nothing :) and since that issue is still open, I can't take advantage of that yet. Should I keep this logic in to prevent the obvious case and then update once the CLR support for ContainsReference comes online?